### PR TITLE
Add wait flag to `scale nodegroup`

### LINKED ISSUE #6396: Support waiting for a `scale nodegroup` operation to complete
eksctl supports a `scale nodegroup` command but the command returns immediately after it has modified the ASG with the new desired size. A `--wait` flag will allow users to wait for the scaling operation to either complete or fail, and monitor any issues related to the scaling. Additionally, a `--timeout` option should also be added to control the maximum duration the command should wait. 

This will also help catch errors like https://github.com/weaveworks/eksctl/issues/6319.

### PR DESCRIPTION
**Pull request by yuxiang-zhang**
### Description

This PR partly covers #6396, i.e. implements functionality for waiting while scaling up. For now, if scaling down, the flag has no effect. We shall decide on what we want to wait when downscaling and open a separate PR.

- Added a wait flag for scale nodegroup command
- Add support for wait scaling managed nodegroups
  - Added an if condition since waiting has been implemented
- Add support for wait scaling unmanaged nodegroups (ASGs)
  - Reused `WaitForNodes` method for scaling up
  - [ ] Implement wait for scaling down

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Comment by yuxiang-zhang** [comment]
I was trying to look for ways to monitor scaling down nodegroups and realized that this is related: 
https://github.com/weaveworks/eksctl/blob/9887e7fca20e3e0d2f79332224be7b8648d13ca8/userdocs/src/usage/managing-nodegroups.md?plain=1#L159

So the idea would be to drain the nodegroups rather than simply updating the ASGs. It seems like a bigger change than what this feature request is about. I'm not sure if we want to address this in this PR? Any thoughts?

**Comment by yuxiang-zhang** [comment]
Manual tests
```sh
λ eksctl get nodegroup -c riki
CLUSTER	NODEGROUP	STATUS		CREATED			MIN SIZE	MAX SIZE	DESIRED CAPACITY	INSTANCE TYPE	IMAGE ID		ASG NAMETYPE
riki	io		CREATE_COMPLETE	2023-03-10T21:06:16Z	1		1		1			m5.large	ami-054178531390b83ba	eksctl-riki-nodegroup-io	unmanaged
riki	lion		ACTIVE		2023-03-10T20:48:05Z	1		1		1			m5.large	AL2_x86_64		eks-lion-063a90370160a3854b1c664c3ec01f1f	managed

λ ./eksctl scale nodegroup -c riki io --nodes-min 2 --nodes-max 3 --wait
2023-03-14 06:53:28 [ℹ]  scaling nodegroup "io" in cluster riki
2023-03-14 06:53:29 [ℹ]  initiated scaling of nodegroup
2023-03-14 06:53:29 [ℹ]  nodegroup "io" has 1 node(s)
2023-03-14 06:53:29 [ℹ]  node "ip-192-168-94-162.us-west-2.compute.internal" is ready
2023-03-14 06:53:29 [ℹ]  waiting for at least 2 node(s) to become ready in "io"
2023-03-14 06:54:47 [ℹ]  nodegroup "io" has 2 node(s)
2023-03-14 06:54:47 [ℹ]  node "ip-192-168-34-99.us-west-2.compute.internal" is ready
2023-03-14 06:54:47 [ℹ]  node "ip-192-168-94-162.us-west-2.compute.internal" is ready

λ ./eksctl scale nodegroup -c riki lion --nodes-min 2 --nodes-max 3 --nodes 2 --wait
2023-03-14 06:55:53 [ℹ]  scaling nodegroup "lion" in cluster riki
2023-03-14 06:55:54 [ℹ]  initiated scaling of nodegroup
2023-03-14 06:55:54 [ℹ]  waiting for scaling of nodegroup "lion" to complete
2023-03-14 06:56:24 [ℹ]  nodegroup successfully scaled

λ eksctl get nodegroup -c riki
CLUSTER	NODEGROUP	STATUS		CREATED			MIN SIZE	MAX SIZE	DESIRED CAPACITY	INSTANCE TYPE	IMAGE ID		ASG NAMETYPE
riki	io		CREATE_COMPLETE	2023-03-10T21:06:16Z	2		3		2			m5.large	ami-054178531390b83ba	eksctl-riki-nodegroup-io	unmanaged
riki	lion		ACTIVE		2023-03-10T20:48:05Z	2		3		2			m5.large	AL2_x86_64		eks-lion-063a90370160a3854b1c664c3ec01f1f	managed

λ ./eksctl scale nodegroup -c riki lion --nodes-min 1 --nodes-max 1 --nodes 1
2023-03-14 07:00:52 [ℹ]  scaling nodegroup "lion" in cluster riki
2023-03-14 07:00:52 [ℹ]  initiated scaling of nodegroup
2023-03-14 07:00:52 [ℹ]  to see the status of the scaling run `eksctl get nodegroup --cluster riki --region us-west-2 --name lion`

λ ./eksctl scale nodegroup -c riki io --nodes-min 1 --nodes-max 1 --nodes 1
2023-03-14 07:02:13 [ℹ]  scaling nodegroup "io" in cluster riki
2023-03-14 07:02:14 [ℹ]  initiated scaling of nodegroup
2023-03-14 07:02:14 [ℹ]  to see the status of the scaling run `eksctl get nodegroup --cluster riki --region us-west-2 --name io`
```

**Comment by TiberiuGC** [comment]
> I was trying to look for ways to monitor scaling down nodegroups and realized that this is related:
> 
> https://github.com/weaveworks/eksctl/blob/9887e7fca20e3e0d2f79332224be7b8648d13ca8/userdocs/src/usage/managing-nodegroups.md?plain=1#L159
> 
> So the idea would be to drain the nodegroups rather than simply updating the ASGs. It seems like a bigger change than what this feature request is about. I'm not sure if we want to address this in this PR? Any thoughts?

Yeah it looks like this needs extra thought and should be handled in a separate ticket. We'd have to discuss this internally first.

File: pkg/actions/nodegroup/scale.go [hunk 1]
```diff
@@ -77,50 +77,64 @@ func (m *Manager) scaleUnmanagedNodeGroup(ctx context.Context, ng *api.NodeGroup
...
+	logger.Info("initiated scaling of nodegroup")
+
+	if wait {
+		timeoutCtx, cancel := context.WithTimeout(ctx, m.ctl.AWSProvider.WaitTimeout())
+		defer cancel()
+		if err := eks.WaitForNodes(timeoutCtx, m.clientSet, ng); err != nil {
+			return err
+		}
 
+		logger.Info("nodegroup successfully scaled")
```
Comment (TiberiuGC) [inline pkg/actions/nodegroup/scale.go:92]: Given that we're probably not going to drain node(s) when downscaling in this PR, I'm guessing there is no effect when using the wait flag when downscaling, is this right? 🤔

**Comment by TiberiuGC** [comment]
Can [this test](https://github.com/weaveworks/eksctl/blob/main/pkg/ctl/drain/nodegroup_test.go) be amended to account for the wait flag?

File: pkg/actions/nodegroup/scale.go (hunk 1 above)
Comment (yuxiang-zhang) [reply pkg/actions/nodegroup/scale.go:92]: Yes, that's the behaviour right now

**Comment by yuxiang-zhang** [comment]
> Can [this test](https://github.com/weaveworks/eksctl/blob/main/pkg/ctl/drain/nodegroup_test.go) be amended to account for the wait flag?

Did you mean this https://github.com/weaveworks/eksctl/blob/main/pkg/ctl/scale/nodegroup_test.go instead? 🤔

**Comment by TiberiuGC** [comment]
> > Can [this test](https://github.com/weaveworks/eksctl/blob/main/pkg/ctl/drain/nodegroup_test.go) be amended to account for the wait flag?
> 
> Did you mean this https://github.com/weaveworks/eksctl/blob/main/pkg/ctl/scale/nodegroup_test.go instead? 🤔

Wrong link yes 🙈

**Comment by yuxiang-zhang** [comment]
@TiberiuGC I'm not sure if this is enough, but I don't see any other test cases to add. Will you please take a look? Thanks! 🙇‍♂️

**Comment by TiberiuGC** [comment]
@yuxiang-zhang apologies for the slow reply. The PR looks generally good. The only concern was that having inconsistent behaviour between scaling up and scaling down (i.e. essentially using the flag on the same comand results in different behaviour) may be confusing from a UX perspective 🤔 

I was wondering if at the very least we could check what type of scaling we're doing and print a warning when scaling down saying that the flag has no effect for now in this scenario. This way we could go ahead and merge the PR as is then figure out the rest. I'll share my thoughts with the team tomorrow.

**Review by TiberiuGC — APPROVED** [review]
LGTM! 👍

**Review by A-Hilaly — APPROVED** [review]
Good stuff! :)

**Pull request merged**